salt/modules/osquery.py

```diff
@@ -56,7 +56,7 @@ def _osquery(sql, format='json'):
     }
 
     cmd = 'osqueryi --json "{0}"'.format(sql)
-    res = __salt__['cmd.run_all'](cmd)
+    res = __salt__['cmd.run_all'](cmd, python_shell=True)
```

Comment:
We want to be careful about command injection here. We want to avoid somebody being able to pass in something that can close the double-quotes and then inject additional command-line arguments which are unrelated to osquery. Something like shlex might be a good tool here.